reframe/frontend/statistics.py

```diff
@@ -197,6 +199,42 @@ def json(self, force=False):
 
         return self._run_data
 
+    def junit(self, json_report, force=False):
+        # https://github.com/windyroad/JUnit-Schema/blob/master/JUnit.xsd
+        xml_testsuites = ET.Element('testsuites')
+        xml_testsuite = ET.SubElement(
+            xml_testsuites, 'testsuite',
+            attrib={
+                'name': 'rfm',
+                'errors': '0',
+                'failures': str(json_report['session_info']['num_failures']),
+                'tests': str(json_report['session_info']['num_cases']),
+                'time': str(json_report['session_info']['time_elapsed']),
+                'hostname': json_report['session_info']['hostname'],
+            }
+        )
+
+        for testid in range(len(json_report['runs'][0]['testcases'])):
+            tid = json_report['runs'][0]['testcases'][testid]
+            name = (
+                f"{tid['name']} on {tid['system']} using {tid['environment']}"
+            )
+            testcase = ET.SubElement(
+                xml_testsuite, 'testcase',
+                attrib={
+                    'classname': tid['filename'],
+                    'name': name,
+                    'time': str(tid['time_total']),
+                }
+            )
+            if not tid['result'] == 'success':
```

Comment:
```suggestion
            if tid['result'] == 'failure':
```

And I'm not sure if we can report `error`. I think yes, in which case you should create a separate entry afaics in the schema.

Reply:
If you click on `View details` in [c4k](https://git.cscs.ch/c4k/c4kcust/example-junit/-/pipelines/23916/test_report), it shows:

> sanity error: pattern 'x' not found in 'rfm_V100_Test_1_job.out'

Do you want more ?

Reply:
Yes, I can see this, but I would like to see other type of failures as well: build failures, performance failures. How do they show up?

Reply:
And this one we need to verify it before merging.

Reply:
I checked this, we have "success," "failure" and "aborted". So the check about "failure" here is correct.